crates/bevy_asset/macros/src/lib.rs

```diff
@@ -12,9 +12,10 @@ pub(crate) fn bevy_asset_path() -> Path {
 }
 
 const DEPENDENCY_ATTRIBUTE: &str = "dependency";
+const ASSET_STORAGE_ATTRIBUTE: &str = "asset_storage";
 
 /// Implement the `Asset` trait.
-#[proc_macro_derive(Asset, attributes(dependency))]
+#[proc_macro_derive(Asset, attributes(dependency, asset_storage))]
```

Comment:
I'd personally prefer to hold off on asset storage changes until we've sorted out "assets as entities". I think we should try to reconcile that design with the "arc-ed assets" needs.